### PR TITLE
Plugin menu

### PR DESCRIPTION
**Pull request by rofe**

### CODE REVIEW

**Comment by rofe** [comment]
@dylandepass I think this would now finally be ready for review, but I can't figure out why some tests are still failing: locally, everything runs through smoothly. Also, codecov seems to be confused from me rebasing this branch multiple times (something I will definitely not repeat ;). In the local coverage report everything new or changes seems covered.

Failing tests which succeed locally:
- https://github.com/adobe/aem-sidekick/blob/plugin-menu/test/app/store/app.test.js#L1242
- https://github.com/adobe/aem-sidekick/blob/plugin-menu/test/app/plugins/delete/delete.test.js#L113

Maybe you can figure it out?

**Review by dylandepass — APPROVED** [review]
Nicely done!

File: src/extension/app/components/plugin/plugin-list/plugin-list.js [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+    // flip the pinned state
+    const newPinnedState = !plugin.isPinned();
+    pluginPrefs.pinned = newPinnedState;
+
+    // toggle button state
+    button.setAttribute('title', newPinnedState
+      ? appStore.i18n('plugin_unpin')
+      : appStore.i18n('plugin_pin'));
+    const icon = button.querySelector('sp-icon');
+    icon.innerHTML = (newPinnedState ? ICONS.STAR_FULL : ICONS.STAR_EMPTY).strings.join('');
```
[the remark this answers is not in the document]
Comment (rofe) [reply src/extension/app/components/plugin/plugin-list/plugin-list.js:125]: Yeah, I know, it's still a workaround 😉

File: src/extension/app/components/plugin/plugin-action-bar.js [hunk 2]
```diff
@@ -97,34 +88,78 @@ export class PluginActionBar extends MobxLitElement {
...
+   * Creates an ESC handler to close the plugin list modal and deselect the toggle button.
+   * @returns {EventListenerOrEventListenerObject} The escape handler
+   */
+  createEscapeHandler() {
+    /**
+     * @param {KeyboardEvent} event The event
+     */
+    return ({ key }) => {
+      if (key === 'Escape') {
+        this.shadowRoot.querySelector('.plugin-list').removeAttribute('selected');
```
[the remark this answers is not in the document]
Comment (rofe) [reply src/extension/app/components/plugin/plugin-action-bar.js]: Ack, but this code is no longer needed.

File: src/extension/app/components/plugin/plugin-list/plugin-list.js [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+        this.requestUpdate();
+      },
+    );
+
+    this.ready = true;
+  }
+
+  async firstUpdated() {
+    // a timeout is needed here to enable keyboard access
+    window.setTimeout(async () => {
```
[the remark this answers is not in the document]
Comment (rofe) [reply src/extension/app/components/plugin/plugin-list/plugin-list.js:100]: Hmm, you are right. Strange, I don't remember why I though it's needed...

**Review by dylandepass — APPROVED** [review]
@rofe I asked killian about the underlay and he didn't want it.. The good news is that if you set the `background-color `of the underlay to `transparent` it still works. So I think we can do that and still have the same experience. Fine with that change being in a future PR.

He also confirmed the button should not be a toggle. So perfect that you removed that.

**Pull request merged**